src/maybe.js

```diff
@@ -0,0 +1,36 @@
+var _curry3 = require('./_curry3');
+
+
+module.exports = (function() {
+  function returnThis() { return this; }
+
+  function Just(x) { this.value = x; }
+  Just.prototype = {
+    of: function _justof(x) { return new Just(x); },
+    map: function _justmap(f) {
+      return new Just(f(this.value));
+    },
+    ap: function _justap(m) {
+      return m.map(this.value);
+    },
+    chain: function _justchain(f) {
+      return f(this.value);
+    },
+    isNothing: false,
+    isJust: true
+  };
+
+  function Nothing() {}
+  Nothing.prototype = {
+    of: Just.prototype.of,
+    map: returnThis,
+    ap: returnThis,
+    chain: returnThis,
+    isNothing: true,
+    isJust: false
+  };
```

Comment:
certainly -- but since this is a strawman, i want to wait and see what direction this whole thing is going to take before buttoning down the details. That's also why the test "suite" is so sparse.

also, i am a bit squeamish about magic string typing; but i can see the case for S compatibility.